### PR TITLE
Support custom types in the checker

### LINKED ISSUE #73: Support custom types in monitor executor
Add support for `enum` and `struct` types

### PR DESCRIPTION
**Pull request by thpani**
Add support for custom types (`struct`, `enum`) in the checker.
The JSDoc on `tlaJsonOfNative()` [explains how the mapping to Apalache types works](https://github.com/freespek/solarkraft/blob/983c4500980107897a773ab2cdf0e4b0fc933203/solarkraft/src/instrument.ts#L121-L139) in detail.

Adds an E2E test, with transactions pre-fetched from the [setter contract](https://github.com/freespek/solarkraft/tree/983c4500980107897a773ab2cdf0e4b0fc933203/ContractExamples/contracts/setter).

Also, extends the setter contract with an `init` function, so we can test the checker with fully initialized fields.

Closes #73

### CODE REVIEW

File: solarkraft/src/instrument.ts [hunk 1]
```diff
@@ -89,30 +105,107 @@ export function tlaJsonTypeOfPrimitive(v: any) {
...
+            // a JS array
+            if (
+                v.length == 0 ||
+                forceVec ||
+                v.every((elem) => typeof elem === typeof v[0])
+            ) {
+                // a Soroban `Vec`
+                // [ 1, 2, 3 ]  ~~>  << 1, 2, 3 >>
+                return {
+                    type: 'Untyped',
```
Comment (konnov) [inline solarkraft/src/instrument.ts:153]: I guess, this will lead to a type ambiguity on an empty sequence

File: solarkraft/test/e2e/tla/.stor/setter/1741597/entry-df673ab5399dcbd5ac434c0ac51c53e9b9367b1d2507ab7e4b8cebfe3e796499.json [hunk 2]
```diff
@@ -0,0 +1 @@
...":"df673ab5399dcbd5ac434c0ac51c53e9b9367b1d2507ab7e4b8cebfe3e796499","contractId":"CBOAR2NNJP77RBSZ7XHTGZGX3BRYGLDUNFRD2CPL5FRKIBBU77XYJCMD","method":"set_bytes","methodArgs":[{"type":"Buffer","data":[190,239]}],"returnValue":null,"fields":[["MY_ADDR","CCLCX5GS3NRUHDLKFBTHWT5CD5US5BUNWCX6H2BN3RKXMVUZMYRXJLBT"],["MY_BOOL",true],["MY_BTES32",{"type":"Buffer","data":[253,237,63,85,222,196,114,80,165,42,140,11,183,3,142,114,250,111,250,174,51,86,47,119,205,43,98,158,247,253,66,77]}],["MY_BYTES",{"type":"Buffer","data":[190,239]}],["MY_ENUM",["A",0]],["MY_I128",-42],["MY_I32",-42],["MY_I64",-42],["MY_MAP",{}],["MY_STRUCT",{"a":0,"b":0}],["MY_SYM","hello"],["MY_U128",42],["MY_U32",42],["MY_U64",42],["MY_VEC",[]]],"oldFields":[["MY_ADDR","CCLCX5GS3NRUHDLKFBTHWT5CD5US5BUNWCX6H2BN3RKXMVUZMYRXJLBT"],["MY_BOOL",true],["MY_BTES32",{"type":"Buffer","data":[253,237,63,85,222,196,114,80,165,42,140,11,183,3,142,114,250,111,250,174,51,86,47,119,205,43,98,158,247,253,66,77]}],["MY_BYTES",{"type":"Buffer","data":[0]}],["MY_ENUM",["A",0]],["MY_I128",-42],["MY_I32",-42],["MY_I64",-42],["MY_MAP",{}],["MY_STRUCT",{"a":0,"b":0}],["MY_SYM","hello"],["MY_U128",42],["MY_U32",42],["MY_U64",42],["MY_VEC",[]]]}
```
Comment (konnov) [inline solarkraft/test/e2e/tla/.stor/setter/1741597/entry-df673ab5399dcbd5ac434c0ac51c53e9b9367b1d2507ab7e4b8cebfe3e796499.json:1]: there is a way to add these files to a github ignore list, so they are collapsed by default

**Review by konnov — APPROVED** [review]
That's great! In the future, we could introduce a mapping file, so we don't have to rely on the type matching heuristics.

File: ContractExamples/scripts/setter-populate.sh [hunk 3]
```diff
@@ -27,6 +27,8 @@ soroban contract build
 soroban contract deploy --wasm target/wasm32-unknown-unknown/release/setter.wasm \
       --source $ACCOUNT --network $NET | tee >.setter.id
 
+soroban contract invoke --id $(cat .setter.id) --source $ACCOUNT --network $NET -- init
```
Comment (Kukovec) [inline ContractExamples/scripts/setter-populate.sh:30]: Why is this necessary?  All of the others overwrite these fields anyway.

File: solarkraft/src/instrument.ts [hunk 4]
```diff
@@ -89,30 +105,107 @@ export function tlaJsonTypeOfPrimitive(v: any) {
...
+                    kind: 'OperEx',
+                    oper: 'TUPLE',
+                    args: v.map((arg) => tlaJsonOfNative(arg)),
```
Comment (Kukovec) [inline solarkraft/src/instrument.ts:156]:
```suggestion
                    args: v.map(tlaJsonOfNative),
```

There's a couple of these, but this is 100% just a style preference, feel free to ignore.

File: solarkraft/test/e2e/verify.test.ts [hunk 5]
```diff
@@ -68,11 +68,39 @@ describe('verify', () => {
...
+        '1bd53234a4eb4f9316f63bd77f4a2c191bded377ad3c53caeb7ed285f9d77d64',
+        '406d278860b5531dd1443532f3457c5daa288e8eb0007d2a8e2aa0127e87949e',
+    ]
+    txs.forEach(tx => {
+        it(`verifies the setter contract (tx ${tx})`, function (done) {
+            this.timeout(150_000)
+            spawn(
+                `solarkraft verify --home test/e2e/tla/ --txHash ${tx} --monitor test/e2e/tla/setter_mon.tla`
+            )
+                .wait('[OK]')
```
Comment (Kukovec) [inline solarkraft/test/e2e/verify.test.ts:102]: is it `[OK]` or `OK`? Because I get errors when running the current tests implying this.

File: solarkraft/test/unit/instrument.test.ts [hunk 6]
```diff
@@ -141,7 +192,18 @@ describe('Apalache JSON instrumentor', () => {
...
+    })
+
+    it('decodes Soroban struct values to Apalache JSON IR', () => {
+        assertProperStruct({ some_long_name: 1 })
+        assertProperStruct({ a: 2, b: 3 })
+    })
+
+    it('decodes Soroban enum values to Apalache JSON IR', () => {
+        assertProperEnum(['A', 1])
+        assertProperEnum(['B', 3, 44])
```
Comment (Kukovec) [inline solarkraft/test/unit/instrument.test.ts:206]: Can you add some negative tests, i.e. expecting throws?

**Review by Kukovec — APPROVED** [review]
LGTM overall

File: solarkraft/test/unit/instrument.test.ts (hunk 6 above)
Comment (thpani) [reply solarkraft/test/unit/instrument.test.ts:206]: You mean for the `assert(false)`?
It's literally meant to be unreachable, so not sure how you'd trigger it?

File: solarkraft/src/instrument.ts (hunk 4 above)
Comment (Kukovec) [reply solarkraft/src/instrument.ts:156]: Ah, I just saw the change from
```
            // a JS array / Soroban `Vec`
            // [ 1, 2, 3 ]  ~~>  << 1, 2, 3 >>
            return {
                type: 'Untyped',
                kind: 'OperEx',
                oper: 'TUPLE',
                args: v.map(tlaJsonOfNative),
```
and I missed that you added a param to the method.

File: solarkraft/test/e2e/verify.test.ts (hunk 5 above)
Comment (Kukovec) [reply solarkraft/test/e2e/verify.test.ts:102]: Screenshot 2024-05-23 at 15 05 32

File: solarkraft/test/unit/instrument.test.ts (hunk 6 above)
Comment (Kukovec) [reply solarkraft/test/unit/instrument.test.ts:206]: No, I mean passing junk inputs. `tlaJsonOfNative` has an out-of-if return which makes `SetAsFun`. However, this should somehow break if it tries to construct e.g. `SetAsFun({ <<"2", 3>>, <<"4", "5">>})`, since that would be an illegal expression for apalache. Right now your test all check for "good" inputs.
or things like `assertProperEnum` asserting false when applied to e.g. a primitive int.

File: solarkraft/test/e2e/verify.test.ts (hunk 5 above)
Comment (Kukovec) [reply solarkraft/test/e2e/verify.test.ts:102]: I fixed this, but the tests still don't succeed:
<details>
<summary>Output</summary>
```
  list
    ✔ lists a single entry (112ms)

  verify
    ✔ errors on missing monitor (107ms)
    ✔ errors on no monitors given (107ms)
    ✔ errors on missing tx hash (110ms)
    1) reports success on okay monitor1
    2) reports success on okay monitor2
    3) reports success on two okay monitors
    4) reports failure on deadlocking monitor
    5) reports failure on deadlocking monitor, if another succeeding monitor is present
    6) verifies the setter contract (tx df32f09514abedbf9d4b843fb8ad53de81d74b62914c4c9faf290760324abd0f)
    7) verifies the setter contract (tx 8f97b05ce36b9f2b2988150131cd8990bd018dfcace5250ad5df5e872af436bf)
    8) verifies the setter contract (tx 03a01ff64502f5d992bb28e784212235a5da55ebb68c75345f58364559cb5921)
    9) verifies the setter contract (tx 337365516cbe5f9287bc6b81428fb94be08d8e5d1ba580454d0907378c06c0da)
    10) verifies the setter contract (tx ad3c4d82300b50972fdbc887c20bb7771c722a1c63bd44c8e8d4ddf4597f4143)
    11) verifies the setter contract (tx e3dba60ada9fd7cb1d3490cf72b5e53e17b154e237aed2d2dcacfec6b0b1d883)
    12) verifies the setter contract (tx 762575ca214bd9cb85824356b799689bf4920aafcac24f790defac54b1a0cc9d)
    13) verifies the setter contract (tx 98c2468aad6626b59230b4fcc83d921897cbf76dc49f7bea13e46e443035a682)
    14) verifies the setter contract (tx df673ab5399dcbd5ac434c0ac51c53e9b9367b1d2507ab7e4b8cebfe3e796499)
    15) verifies the setter contract (tx 101111c4a7087070366d035f59cc20b80215231955f9253c4ad85ff94f1dc61b)
    16) verifies the setter contract (tx 9c46c29a985ceeaffe567dc405a757805ff0496fe5740228abcb22b7aefa3e64)
    17) verifies the setter contract (tx 4da2e37effde690b6cf2c11ade5fc2f4e72971b37866f6d75fa2ea5728b6be6e)
    18) verifies the setter contract (tx b39cf7d57120fa72bde0aec90f709551ff777132e1b5db89e2575ee942d8bb70)
    19) verifies the setter contract (tx 1bd53234a4eb4f9316f63bd77f4a2c191bded377ad3c53caeb7ed285f9d77d64)
    20) verifies the setter contract (tx 406d278860b5531dd1443532f3457c5daa288e8eb0007d2a8e2aa0127e87949e)


  4 passing (3s)
  20 failing

  1) verify
       reports success on okay monitor1:

      AssertionError [ERR_ASSERTION]: expected "timelock_mon1.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -timelock_mon1.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  2) verify
       reports success on okay monitor2:

      AssertionError [ERR_ASSERTION]: expected "timelock_mon2.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -timelock_mon2.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  3) verify
       reports success on two okay monitors:

      AssertionError [ERR_ASSERTION]: expected "timelock_mon2.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -timelock_mon2.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  4) verify
       reports failure on deadlocking monitor:

      AssertionError [ERR_ASSERTION]: Non-empty queue on spawn exit.
[wait] unable to reproduce the transaction
[wait] [Fail]
      + expected - actual

      -[
      -  "[wait] unable to reproduce the transaction"
      -  "[wait] [Fail]"
      -]
      +[]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createUnexpectedEndError (node_modules/nexpect/lib/nexpect.js:322:10)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:225:19)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  5) verify
       reports failure on deadlocking monitor, if another succeeding monitor is present:

      AssertionError [ERR_ASSERTION]: expected "timelock_mon2.tla: Internal error, see below" to contain "[Fail]"
      + expected - actual

      -timelock_mon2.tla: Internal error, see below
      +[Fail]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  6) verify
       verifies the setter contract (tx df32f09514abedbf9d4b843fb8ad53de81d74b62914c4c9faf290760324abd0f):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  7) verify
       verifies the setter contract (tx 8f97b05ce36b9f2b2988150131cd8990bd018dfcace5250ad5df5e872af436bf):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  8) verify
       verifies the setter contract (tx 03a01ff64502f5d992bb28e784212235a5da55ebb68c75345f58364559cb5921):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  9) verify
       verifies the setter contract (tx 337365516cbe5f9287bc6b81428fb94be08d8e5d1ba580454d0907378c06c0da):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  10) verify
       verifies the setter contract (tx ad3c4d82300b50972fdbc887c20bb7771c722a1c63bd44c8e8d4ddf4597f4143):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  11) verify
       verifies the setter contract (tx e3dba60ada9fd7cb1d3490cf72b5e53e17b154e237aed2d2dcacfec6b0b1d883):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  12) verify
       verifies the setter contract (tx 762575ca214bd9cb85824356b799689bf4920aafcac24f790defac54b1a0cc9d):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  13) verify
       verifies the setter contract (tx 98c2468aad6626b59230b4fcc83d921897cbf76dc49f7bea13e46e443035a682):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  14) verify
       verifies the setter contract (tx df673ab5399dcbd5ac434c0ac51c53e9b9367b1d2507ab7e4b8cebfe3e796499):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  15) verify
       verifies the setter contract (tx 101111c4a7087070366d035f59cc20b80215231955f9253c4ad85ff94f1dc61b):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  16) verify
       verifies the setter contract (tx 9c46c29a985ceeaffe567dc405a757805ff0496fe5740228abcb22b7aefa3e64):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  17) verify
       verifies the setter contract (tx 4da2e37effde690b6cf2c11ade5fc2f4e72971b37866f6d75fa2ea5728b6be6e):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  18) verify
       verifies the setter contract (tx b39cf7d57120fa72bde0aec90f709551ff777132e1b5db89e2575ee942d8bb70):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  19) verify
       verifies the setter contract (tx 1bd53234a4eb4f9316f63bd77f4a2c191bded377ad3c53caeb7ed285f9d77d64):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  20) verify
       verifies the setter contract (tx 406d278860b5531dd1443532f3457c5daa288e8eb0007d2a8e2aa0127e87949e):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)
```
</details>

Comment (konnov) [reply solarkraft/test/e2e/verify.test.ts:102]: It's also failing for me. I guess, it's only working in the dev container?

File: solarkraft/test/unit/instrument.test.ts (hunk 6 above)
Comment (konnov) [reply solarkraft/test/unit/instrument.test.ts:206]: I would say that we could write tests for the negative inputs earlier, but not now. We can make a separate issue on this.

File: solarkraft/test/e2e/verify.test.ts (hunk 5 above)
Comment (thpani) [reply solarkraft/test/e2e/verify.test.ts:102]: @Kukovec I think you'll have to run `verify` manually to see why it fails.
E.g., do you have Apalache at the expected path?

File: solarkraft/test/unit/instrument.test.ts (hunk 6 above)
Comment (thpani) [reply solarkraft/test/unit/instrument.test.ts:206]: > Right now your test all check for "good" inputs.

They check for inputs that are possible from Soroban contracts. If you have an example of plausible input that isn't covered, please open an issue.

File: ContractExamples/scripts/setter-populate.sh (hunk 3 above)
Comment (thpani) [reply ContractExamples/scripts/setter-populate.sh:30]: As I explained in the call yesterday, unassigned contract fields will cause `apalache-mc check` to fail, so you want to make sure that all fields exist from the beginning.

File: solarkraft/test/e2e/tla/.stor/setter/1741597/entry-df673ab5399dcbd5ac434c0ac51c53e9b9367b1d2507ab7e4b8cebfe3e796499.json (hunk 2 above)
Comment (thpani) [reply solarkraft/test/e2e/tla/.stor/setter/1741597/entry-df673ab5399dcbd5ac434c0ac51c53e9b9367b1d2507ab7e4b8cebfe3e796499.json:1]: Oh true, that's amazing!
https://github.com/github-linguist/linguist/blob/master/docs/overrides.md#generated-code

File: solarkraft/src/instrument.ts (hunk 1 above)
Comment (thpani) [reply solarkraft/src/instrument.ts:153]: Hmmm, I'm not sure. Right now this will be assigned to the respective variable, so as long as that variable is typed correctly in the monitor, I think it is fine?

Anyhow, in general we could use more type info here, so I'll leave this comment unresolved so we can find it later on.

File: solarkraft/test/e2e/verify.test.ts (hunk 5 above)
Comment (thpani) [reply solarkraft/test/e2e/verify.test.ts:102]: I assume this was resolved by #79?
